frontend/packages/console-dynamic-plugin-sdk/src/runtime/plugin-dependencies.ts

```diff
@@ -70,7 +70,12 @@ export const resolvePluginDependencies = (
 
   // Ensure compatibility with other dynamic plugins
   const requiredPluginNames = _.difference(Object.keys(dependencies), [pluginAPIDepName]);
-  const unavailablePluginNames = _.difference(requiredPluginNames, allowedPluginNames);
+
+  const unavailablePluginNames = _.difference(
+    requiredPluginNames,
+    allowedPluginNames,
+    manifest.customProperties?.console?.optionalDependencies ?? [],
```

Comment:
what happens if a plugin is in `optionalDependencies` but is not in `dependencies`? should this be validated against?

Reply:
@logonoff This kind of inconsistency is currently ignored.

I'll look into moving to `optionalDependencies` object approach as you've suggested.